lib/like-pattern-concordance.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const concordance = require('concordance');
+const isPlainObject = require('is-plain-object');
+
+function pickLikePatternPaths(actual, likePattern, visited = new WeakMap()) {
+	if (visited.has(likePattern)) {
+		return visited.get(likePattern);
+	}
+
+	if (isPlainObject(likePattern) && isPlainObject(actual)) {
+		const actualPicked = {};
+
+		visited.set(likePattern, actualPicked);
+
+		for (const key of Object.keys(likePattern)) {
+			if (key in actual) {
+				actualPicked[key] = pickLikePatternPaths(actual[key], likePattern[key], visited);
```

Comment:
This is probably not an issue, but Concordance avoids recursion, whereas this will eventually hit stack limits.

Reply:
I have [a test for circular pattern](https://github.com/avajs/ava/pull/2490/files#diff-7ebbd1d7d5643b04618ce7d1c59ce9b8R857). I don't have a test for more-than-stack-limit-deep pattern though. But patterns are expected to be written by hand so I hope this is not an issue too.